keras/preprocessing/image.py

```diff
@@ -1104,7 +1104,7 @@ def _get_batches_of_transformed_samples(self, index_array):
                 img = array_to_img(batch_x[i], self.data_format, scale=True)
                 fname = '{prefix}_{index}_{hash}.{format}'.format(prefix=self.save_prefix,
                                                                   index=j,
-                                                                  hash=np.random.randint(1e4),
+                                                                  hash=np.random.randint(2**31 - 1),
```

Comment:
Since Python is a compiled language you are recomputing `2 ** 31` at every iteration here. Prefer instead using e.g. 1e9 or something. 1e7 would be enough I guess. But really I'm not sure this is a good idea since it makes the file names less readable.

Reply:
1e7 would be good. Personally, I think the readability of the file name is a minor issue, especially since these files are mainly meant for debugging purposes anyways. I can make the commit to change it to 1e7.